### PR TITLE
Copy new and modified Hyrax feature specs

### LINKED ISSUE #130: Copy newer feature specs from Hyrax
### Descriptive summary

Some feature specs have been added or changed since we last copied specs from Hyrax.  Copy the new tests in spec/features/ and get them passing.

New:
* spec/features/collection_multi_membership_spec.rb
* spec/features/collection_type_spec.rb
* spec/features/dashboard/all_works.rb
* spec/features/dashboard/collection_spec.rb
* spec/features/dashboard/display_dashboard_spec.rb

Changed:
* spec/features/collection_spec.rb

### PR DESCRIPTION
**Pull request by hortongn**
Fixes #130

This copies feature specs from Hyrax that have been added or modified over the past few weeks.  

These tests make it more likely we will know if our app has broken something in Hyrax.

Since these specs come straight from Hyrax, the reviewer probably doesn't need to examine the code in detail.  I only made very slight tweaks to some of the specs to get them to run properly in our app.

### CODE REVIEW

File: spec/features/dashboard/collection_spec.rb [hunk 1]
```diff
@@ -0,0 +1,877 @@
...
+      end
+
+      it "has page title, has tabs for All and Your Collections, and lists collections with edit access" do
+        expect(page).to have_content 'Collections'
+        expect(page).to have_link 'All Collections'
+        expect(page).to have_link 'Your Collections'
+        expect(page).to have_link(collection3.title.first)
+        expect(page).to have_link(collection4.title.first)
+        expect(page).to have_link(admin_set_a.title.first)
+        expect(page).not_to have_link(collection1.title.first)
```
Comment (jamesvanmil) [inline spec/features/dashboard/collection_spec.rb]: This line is failing for me when running locally via `rake ci`

Comment (jamesvanmil) [inline spec/features/dashboard/collection_spec.rb]: This line is failing for me when running locally via `rake ci`

Comment (hortongn) [reply spec/features/dashboard/collection_spec.rb]: @jamesvanmil In addition to using the seed you provided, I can get the test to fail by running just that test (spec/features/dashboard/collection_spec.rb:88) by itself.  Hyrax (2.1.0.beta1) has the same feature specs and when I run this individual test there it fails as well.  So I'm seeing if there's an obvious reason it's failing.  It must rely on something getting set up in a different block in that file.

**Comment by scherztc** [comment]
My specs on this branch passed locally.
Finished in 5 minutes 0 seconds (files took 9.09 seconds to load)
157 examples, 0 failures, 8 pending

**Pull request merged**